packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -478,8 +478,11 @@ export default class {
   }
 
   // returns the XAPI object corresponding to an XO object
-  getXapiObject(xoObject) {
-    return this.getXapi(xoObject).getObjectByRef(xoObject._xapiRef)
+  getXapiObject(xoObject, type) {
+    if (typeof xoObject === 'string') {
+      xoObject = this._xo.getObject(xoObject, type)
+    }
+    return this.getXapi(xoObject, type).getObjectByRef(xoObject._xapiRef)
```

Comment:
Why do we need to pass `type` to `getXapi`?

Reply:
To check the type of the passed object.

Reply:
I understand asserting the type when passing an id but IMHO, it does not make sense when passing an object.

Reply:
Done :+1: